Automattic/wp-calypso
# Add sites migration banner

Related to #95161

## Proposed Changes

* Add a banner in the `/sites` dashboard to help the user in case they have a pending migration.

## Why are these changes being made?


* To help the user complete their migration in case they are stuck.

## Testing Instructions



* Without any site with pending migration, go to `/sites`, and make sure the new banner doesn't apper.
* Test that with less than 5 sites (test just to make sure it continues working - it's the current behavior just refactored) in the `/sites` you also don't see the agency banner (`Building sites for customers? Here's how to earn more.`).
* Test that with 5 sites or more, you see the banner.
* Now go to `/setup/migration`, and navigate through the flow until select the option "Do it for me". Close the window in the step "Tell us about your WordPress site".
* Go to `/sites`, and check that you see only the migration banner (`Stuck on your migration?`).
* Dismiss it clicking on the "X", and make sure you see the agency banner again.

## Screenshots

<img width="1069" alt="Screenshot 2024-10-10 at 19 18 54" src="https://github.com/user-attachments/assets/d3ae4c6d-bd65-4757-8769-68fdceb69150">

In case you want to undo a dismiss to retest something, the easiest way I thought of was to apply the following diff and click on the "X" (persisting it as not dismissed):

```diff
diff --git a/client/blocks/dismissible-card/actions.js b/client/blocks/dismissible-card/actions.js
index a6e2f6d8d6..94dfc92424 100644
--- a/client/blocks/dismissible-card/actions.js
+++ b/client/blocks/dismissible-card/actions.js
@@ -5,8 +5,8 @@ export const dismissCard = ( preferenceName, temporary ) => {
        const prefKey = getPreferenceKey( preferenceName );
 
        if ( temporary ) {
-               return setPreference( prefKey, true );
+               return setPreference( prefKey, false );
        }
 
-       return savePreference( prefKey, true );
+       return savePreference( prefKey, false );
 };
diff --git a/client/blocks/dismissible-card/index.jsx b/client/blocks/dismissible-card/index.jsx
index f867a143d0..7085ad324a 100644
--- a/client/blocks/dismissible-card/index.jsx
+++ b/client/blocks/dismissible-card/index.jsx
@@ -23,7 +23,7 @@ function DismissibleCard( {
        const dispatch = useDispatch();
        const translate = useTranslate();
 
-       if ( isDismissed || ! hasReceivedPreferences ) {
+       if ( false ) {
                return null;
        }
```

## Pre-merge Checklist
